VMEncryption/main/ResourceDiskUtil.py

```diff
@@ -0,0 +1,274 @@
+
+    def is_encrypt_format_all(self):
+        """ return true if current encryption operation is EncryptFormatAll """
+        try:                
+            public_settings_str = self.hutil._context._config['runtimeSettings'][0]['handlerSettings'].get('publicSettings')
+            if isinstance(public_settings_str, basestring):
+                public_settings = json.loads(public_settings_str)
+            else:
+                public_settings = public_settings_str
+            encryption_operation = public_settings.get(CommonVariables.EncryptionEncryptionOperationKey)
+            if encryption_operation in [CommonVariables.EnableEncryptionFormatAll]:
+                return True
+        except:
+            self.logger.log("unable to identify current encryption operation")
+        return False
+
+    def is_luks_device(self):
+        """ checks if the device is set up with a luks header """
+        if not self.resource_disk_partition_exists():
+            return False
+        cmd = 'cryptsetup isLuks ' + self.RD_DEV_PATH
+        return (int)(self.executor.Execute(cmd, suppress_logging=True)) == 0
+
+    def is_luks_device_opened(self):
+        """ check for presence of luks uuid to see if device was already opened """
+        # suppress logging to avoid log clutter if the device is not open yet
+        if not self.resource_disk_partition_exists():
+            return False
+        cmd = 'test -b /dev/disk/by-uuid/$(cryptsetup luksUUID ' + self.RD_DEV_PATH + ')'
+        return (int)(self.executor.ExecuteInBash(cmd, suppress_logging=True)) == 0
+
+    def is_valid_key(self):
+        """ test if current key can be used to open current partition """
+        # suppress logging to avoid log clutter if the key doesn't match
+        if not self.resource_disk_partition_exists():
+            return False
+        cmd = 'cryptsetup luksOpen ' + self.RD_DEV_PATH + ' --test-passphrase --key-file ' + self.RD_KEY_FILE
+        return (int)(self.executor.Execute(cmd, suppress_logging=True)) == 0
+
+    def resource_disk_exists(self):
+        """ true if the udev name for resourced disk exists """
+        cmd = 'test -b ' + self.RD_BASE_DEV_PATH
+        return (int)(self.executor.Execute(cmd, suppress_logging=True)) == 0
+
+    def resource_disk_partition_exists(self):
+        """ true if udev name for resource disk partition exists """
+        cmd = 'test -b ' + self.RD_DEV_PATH
+        return (int)(self.executor.Execute(cmd, suppress_logging=True)) == 0
+
+    def format_luks(self):
+        """ set up resource disk crypt device layer using disk util """
+        if not self.resource_disk_partition_exists():
+            self.logger.log('LUKS format operation requested, but resource disk partition does not exist')
+            return False
+        return (int)(self.disk_util.luks_format(passphrase_file=self.RD_KEY_FILE, dev_path=self.RD_DEV_PATH, header_file=None)) == 0
```

Comment:
Why not just do ( == CommonVariables.process_success ).
I don't like it any more than you do but there is something to be said for keeping code consistent.

Reply:
OK, will update